tests/multi_ptr/multi_ptr_comparison_op.h

```diff
@@ -0,0 +1,591 @@
+      // result
+      detail::mptr_mptr_test_results<T> expected_results;
+      // Expected value that will be returned after "first_mptr != first_mptr"
+      // operator will be called
+      expected_results.first_first_mptr_result_val = false;
+      // Expected value that will be returned after "first_mptr != second_mptr"
+      // operator will be called
+      expected_results.first_second_mptr_result_val = true;
+
+      const auto run_test_action = [](auto arr_acc, auto result_acc) {
+        multi_ptr_t arr_mptr(arr_acc);
+        // multi_ptr that pointed at the first element
+        multi_ptr_t mptr_1(arr_mptr);
+        // multi_ptr that pointed at the second element
+        multi_ptr_t mptr_2(arr_mptr + 1);
+
+        auto &test_result = result_acc[0];
+
+        test_result.first_first_mptr_result_val = mptr_1 != mptr_1;
+        test_result.first_second_mptr_result_val = mptr_1 != mptr_2;
+      };
+
+      run_test(queue, run_test_action, expected_results);
+    }
+    SECTION(section_name(
+                "Check multi_ptr operator<(const multi_ptr&, const multi_ptr&)")
+                .with("T", type_name)
+                .with("address_space", address_space_name)
+                .with("decorated", is_decorated_name)
+                .create()) {
+      // Variable that contains all variables that will be used to verify test
+      // result
+      detail::mptr_mptr_test_results<T> expected_results;
+      // Expected value that will be returned after "first_mptr < first_mptr"
+      // operator will be called
+      expected_results.first_first_mptr_result_val = false;
+      // Expected value that will be returned after "first_mptr < second_mptr"
+      // operator will be called
+      expected_results.first_second_mptr_result_val = true;
+
+      const auto run_test_action = [](auto arr_acc, auto result_acc) {
+        multi_ptr_t arr_mptr(arr_acc);
+        // multi_ptr that pointed at the first element
+        multi_ptr_t mptr_1(arr_mptr);
+        // multi_ptr that pointed at the second element
+        multi_ptr_t mptr_2(arr_mptr + 1);
+
+        auto &test_result = result_acc[0];
+
+        test_result.first_first_mptr_result_val = mptr_1 < mptr_1;
+        test_result.first_second_mptr_result_val = mptr_1 < mptr_2;
+      };
+
+      run_test(queue, run_test_action, expected_results);
+    }
+    SECTION(section_name(
+                "Check multi_ptr operator>(const multi_ptr&, const multi_ptr&)")
+                .with("T", type_name)
+                .with("address_space", address_space_name)
+                .with("decorated", is_decorated_name)
+                .create()) {
+      // Variable that contains all variables that will be used to verify test
+      // result
+      detail::mptr_mptr_test_results<T> expected_results;
+      // Expected value that will be returned after "first_mptr > second_mptr"
+      // operator will be called
+      expected_results.first_first_mptr_result_val = false;
```

Comment:
```suggestion
      expected_results.first_second_mptr_result_val = false;
```

Reply:
I've applied this comment [here](https://github.com/KhronosGroup/SYCL-CTS/pull/363/commits/90f72c25c6b6e49ffa2ec5179953f7f3eab373a6#diff-6694f76c9f9ec20ae7ad1201253bd9ce5172455eec7b5a61db5b7b800a5ce57dL281-R276)